api/net/Areas/Helpers/WorkOrderHelper.cs

```diff
@@ -0,0 +1,135 @@
+using System.Security.Claims;
+using Microsoft.Extensions.Options;
+using TNO.API.Config;
+using TNO.Core.Exceptions;
+using TNO.Core.Extensions;
+using TNO.DAL.Services;
+using TNO.Kafka;
+
+namespace TNO.API.Helpers;
+
+/// <summary>
+/// WorkOrderHelper class, provides methods to help with work orders.
+/// </summary>
+public class WorkOrderHelper : IWorkOrderHelper
+{
+    #region Variables
+    /// <summary>
+    /// The following work order status ensure only a single request can be completed for content.
+    /// </summary>
+    public static readonly IEnumerable<Entities.WorkOrderStatus> WorkLimiterStatus = new[] { Entities.WorkOrderStatus.Submitted, Entities.WorkOrderStatus.InProgress, Entities.WorkOrderStatus.Completed };
+
+    private readonly ClaimsPrincipal _principal;
+    private readonly IContentService _contentService;
+    private readonly IWorkOrderService _workOrderService;
+    private readonly IUserService _userService;
+    private readonly IKafkaMessenger _kafkaMessenger;
+    private readonly KafkaOptions _kafkaOptions;
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a WorkOrderHelper object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="principal"></param>
+    /// <param name="contentService"></param>
+    /// <param name="workOrderService"></param>
+    /// <param name="userService"></param>
+    /// <param name="kafkaMessenger"></param>
+    /// <param name="kafkaOptions"></param>
+    public WorkOrderHelper(
+        ClaimsPrincipal principal,
+        IContentService contentService,
+        IWorkOrderService workOrderService,
+        IUserService userService,
+        IKafkaMessenger kafkaMessenger,
+        IOptions<KafkaOptions> kafkaOptions)
+    {
+        _principal = principal;
+        _contentService = contentService;
+        _workOrderService = workOrderService;
+        _userService = userService;
+        _kafkaMessenger = kafkaMessenger;
+        _kafkaOptions = kafkaOptions.Value;
+    }
+    #endregion
+
+    #region Methods
+    /// <summary>
+    /// Determine if the content should be auto transcribed.
+    /// </summary>
+    /// <param name="contentId"></param>
+    /// <returns></returns>
+    /// <exception cref="NoContentException"></exception>
+    public bool ShouldAutoTranscribe(long contentId)
+    {
+        var content = _contentService.FindById(contentId);
+        return content?.ContentType == Entities.ContentType.AudioVideo &&
+            content.FileReferences.Any() &&
+            !content.IsApproved &&
+            (content.Source?.AutoTranscribe == true ||
```

Comment:
If content is part of a source, series, or contributor that has been configured to auto transcribe it will transcribe.